proxy/attempt-register-vm-on-proxy.sh

```diff
@@ -35,15 +35,6 @@ function run-proxy-agent {
         --health-check-unhealthy-threshold=${HEALTH_CHECK_UNHEALTHY_THRESHOLD}
 }
 
-# Check if the cluster already have proxy agent installed by checking ConfigMap.
-if kubectl get configmap inverse-proxy-config; then
```

Comment:
I wonder why we abandoned this caching completely. Would that be a problem, like leaking resources?

Reply:
Yes removing the caching will result in a new URL each time proxy agent start for any reason. I think what you could do is instead check if proxy_uro is empty string.

Reply:
The configmap is always there, so we should not check like before.
As for whether instead check if key-value is empty, it will introduce bug and not necessary.

1. why introduce bug
Currently we use VM ID as Key to get/create hostName. If we just check key-value is empty, then it's possible a pod is restarted in a node in another VM and trigger the proxy-agent can't well connect with proxy-server.

2. why not necessary
Please check proxy server side codes. It uses VM ID as key to get/create hostname. It doesn't always create a new hostname. Also, restarting the proxy-agent is not frequent.

Reply:
The code in proxy server uses the VM ID as unique identifier for a GKE cluster since GKE doesn't have one. This is by design. VM ID is no longer used when doing the actual routing.

Without the check the proxy-agent will create a new URL when it fail over to a new VM.

Reply:
Thanks yang. Checked the internal codes. It actually use both VM ID and service account identity, aka we still better let server decide whether get existing or use new. It uses both VM ID (backend hostname) and service account email to query whether existing data there.

I pinged you internal code link (can't paste here).

Reply:
as talked offline, we don't need to worry about VM ID as server side will ignore VM ID.

In the future, if we find moving pod to another VM actually can't work, we go back to delete this check.